core/listener.go

```diff
@@ -211,13 +167,13 @@ func (cl *Listener) listen(ctx context.Context, sub <-chan types.EventDataSigned
 			if !timeout.Stop() {
 				<-timeout.C
 			}
-			timeout.Reset(cl.listenerTimeout)
 		case <-timeout.C:
```

Comment:
can we just track this in fetcher and remove the timer from listener entirely?

Reply:
along with the metric(subscription is stuck)?

Reply:
I understand the point of the metric. I didn't say we should remove the metric - i'm saying we should track the metric in fetcher where it occurs and remove the concept of the `timer` from the listener listen loop.

Reply:
I don't think the fetcher should know about any limitations of waiting for new blocks. IMO, the fetcher should only wait for them and pass to the listener which responsible for the verification, storing and broadcasting them.